snakeviz/static/snakeviz.js

```diff
@@ -1,227 +1,278 @@
 };
 
-
 // Make the worker and sv_draw_vis function
-var sv_make_worker = function sv_make_worker() {
-    var URL = URL || window.URL || window.webkitURL;
-    var blob = new Blob(
-        [$('#heirarchy-worker').text()], {'type': 'text/javascript'});
-    var blobURL = URL.createObjectURL(blob);
-    var sv_worker = new Worker(blobURL);
-
-    sv_worker.onmessage = function (event) {
-        var json = JSON.parse(event.data);
-        if (cache_key != null) {
-            sv_json_cache[cache_key] = json;
-        }
-        redraw_vis(json);
-        _.defer(sv_hide_working);
-    };
-
-    sv_worker.onerror = function (event) {
-        sv_show_error_msg();
-        console.log(event);
-        sv_cycle_worker();
-        sv_hide_working();
-    };
-
-    sv_end_worker = function () {
-        sv_worker.terminate();
-        URL.revokeObjectURL(blobURL);
-        sv_hide_working();
-    };
-
-    return sv_worker;
+var sv_make_worker = function() {
+  var URL = URL || window.URL || window.webkitURL;
+  var blob = new Blob(
+      [$('#heirarchy-worker').text()], {'type': 'text/javascript'});
+  var blobURL = URL.createObjectURL(blob);
+  var sv_worker = new Worker(blobURL);
+  
+  sv_worker.onmessage = function (event) {
+    cache_key = JSON.stringify(event.data);        
+    if (!(_.has(sv_json_cache, cache_key))) {
+    	var json = JSON.parse(event.data);
+      sv_json_cache[cache_key] = json;
+    }
+    clear_and_redraw_vis(sv_json_cache[cache_key]);
+  };
+  
+  sv_worker.onerror = function (event) {
+    sv_show_error_msg();
+    sv_cycle_worker();
+    sv_hide_working();
+  };
+  
+  sv_end_worker = function () {
+    sv_worker.terminate();
+    URL.revokeObjectURL(blobURL);
+    sv_hide_working();
+  };
+  
+  return sv_worker;
 };
 
 
-var sv_cycle_worker = function sv_cycle_worker() {
-    sv_end_worker();
-    sv_worker = sv_make_worker();
+var sv_cycle_worker = function() {
+  sv_end_worker();
+  sv_worker = sv_make_worker();
 };
 
 
-var sv_draw_vis = function sv_draw_vis(root_name, parent_name) {
-    sv_show_working();
-    var message = {
-        'depth': sv_heirarchy_depth(),
-        'cutoff': sv_heirarchy_cutoff(),
-        'name': root_name,
-        'parent_name': parent_name,
-        'url': window.location.origin
-    };
-
-    cache_key = JSON.stringify(message);
-    if (_.has(sv_json_cache, cache_key)) {
-        redraw_vis(sv_json_cache[cache_key]);
-        sv_hide_working();
-    } else {
-        sv_worker.postMessage(message);
-    }
+var sv_draw_vis = function(root_name, parent_name,init) {
+  sv_show_working();
+  if (root_name != null){
+    callStack.updateStack(root_name,parent_name);
+    callStack.checkResetButtonState();
+  }
+  var message = buildMessage(callStack.currentRoot(),callStack.currentParent(),init);
+  sv_worker.postMessage(message);
+  _.defer(sv_hide_working);
 };
 
+var buildMessage = function(rootName, parentName,initialRun){
+	if (initialRun){
+		depth = 10000;
```

Comment:
What the reasoning behind this initial 10000 / 0? That seems likely to lead to some crashes on large profiles.